packages/@lightningjs/ui-components/src/mixins/withThemeStyles/StyleManager.js

```diff
@@ -0,0 +1,295 @@
+/**
+ * Copyright 2023 Comcast Cable Communications Management, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  generateComponentStyleSource,
+  generateStyle,
+  replaceAliasValues
+} from './utils.js';
+import lng from '@lightningjs/core';
+import { debounce } from '../../utils/index.js';
+import { context } from '../../globals/index.js';
+import { clone } from '../../utils/index.js';
+
+export default class StyleManager extends lng.EventEmitter {
+  /**
+   * Create a Style Manager instance.
+   * @param {Object} options - The options object.
+   * @param {Object} options.component - The LightningJS component instance.
+   */
+  constructor({ component = {} } = {}) {
+    super(...arguments);
+    this.resolving = false;
+    this._previousTone;
+    this._previousMode;
+    this._previousComponentLevelStyle;
+    // Bind the _onThemeUpdate method to the current instance of the StyleManager class.
+    this._boundThemeUpdate = this._onThemeUpdate.bind(this);
+    context.on('themeUpdate', this._boundThemeUpdate);
+
+    // Create a global cache for styles.
+    if (!window.LUI_STYLE_CACHE) {
+      window.LUI_STYLE_CACHE = new Map();
+    }
+
+    this.component = component;
+    this._style = {};
+    // Initial Update
+    this._update();
+
+    if (process?.env?.NODE_ENV === 'test') {
+      this.update = this._update; // Avoid race conditions in tests
```

Comment:
can you clarify this at all? It looks like `this.update()` is only ever called here in this file. Why not just change it to the underscore value and remove this?

Reply:
update in place of _update only was put in place to make sure the call the _update is debounced. I could rename it if you think it is more clear however.

Reply:
i think naming it something else would be better, also should it be underscored or is this part of the API?